lib/translations_checker/git_diff.rb

```diff
@@ -0,0 +1,40 @@
+require "translations_checker/concerns/service"
+
+module TranslationsChecker
+  class GitDiff
+    include Concerns::Service
+
+    DEFAULT_OPTIONS = { diff_filter: "AM" }.freeze
+
+    attr_reader :path, :ref
+
+    def initialize(path, ref: "origin/master")
+      @path = path
+      @ref = ref
+    end
+
+    def diff
+      cmd = %w(git diff --diff-filter=AM --unified=0) + %W("#{ref}" "#{path}")
```

Comment:
Maybe we could diff only what is being pushed, not the whole current working directory against `origin/master`.
Currently the diff includes changes made on `origin/master` that are not on our branch and also changes in the working directory that are not being pushed.
What we could do instead is `git diff origin/master...HEAD`, which includes only changes between the common ancestor of (`origin/master`, `HEAD`) and `HEAD`. This is the same thing that Github does when showing changes in the PR. Check [docs](https://git-scm.com/docs/git-diff#git-diff-emgitdiffem--optionsltcommitgtltcommitgt--ltpathgt82308203) for more info.

Reply:
I kind of liked that it was diffing uncommitted changes in the working directory - it makes testing a bit easier. But I agree that diffing only what is being pushed, and against the common ancestor makes more sense.

Reply:
Actually, there is one problem with using `origin/master...HEAD`. When we build the key map we load the working directory copy. Would `git diff master` work?

Reply:
... or I can use `git show HEAD:...`

Reply:
There is no much difference between `git diff master` and `git diff origin/master`, they both can be outdated if you haven't fetched recently. I would suggest to build the key map from `HEAD` directory instead of a working directory. 
So :+1: for `git show HEAD:path/to/file`

Reply:
Done